core/trino-main/src/main/java/io/trino/execution/CallTask.java

```diff
@@ -118,6 +118,9 @@ public String getName()
         Predicate<CallArgument> hasName = argument -> argument.getName().isPresent();
         boolean anyNamed = call.getArguments().stream().anyMatch(hasName);
         boolean allNamed = call.getArguments().stream().allMatch(hasName);
+        if (!allNamed && procedure.isNameArgumentsOnly()) {
```

Comment:
Ok Thx, let me know can we proceed with that. This is limiting spec, but i think in resonable way within its boundries.

Reply:
What's the motivation for this? Why would we ever need it?

Reply:
In case when we have all arguments optional and logic is dependent on set non-required arguments.
This would not cause backward compatibility issues if we will ever decide some new argument is added or required.
Only documentation would need to be update.
Perfect example is current flush_metadata_cache procedure.
For now we allow below use cases
```sql
USE example_catalog.some_schema;

# Clear all caches in catalog 'example_catalog' for all schema
system.flush_metadata_cache(); 

# Clear all caches for specific partition 'partition_column=partition_value'
# in table 'example_catalog.example_schema.example_table'
system.flush_metadata_cache('example_schema', 'example_table', ARRAY['partition_column'], ARRAY['partition_value']);
```
But at some point we might extend scope to
```sql
USE example_catalog.some_schema;

# Clear all caches for specific schema 'example_catalog.example_schema'
system.flush_metadata_cache('example_schema');

# Clear all caches for specific table 'example_catalog.example_schema.example_table'
system.flush_metadata_cache('example_schema', 'example_table');
```
Maybe even other use cases which are hard to predict at the moment. 

CC: @findepi